### ISSUE TITLE
Create hashes using string builder instead of big integers

### ISSUE THREAD

**Issue by KilianB** [KilianB/JImageHash#22]
During hash creation a bigInteger object will be created for each bit present. Currently no fancy operation take place during it's creation process that justify this overhead.
Afterwards the xor operation on arbitrary hashes is required for the hamming distance calculation.

Benchmark if it's more performant to use a stringbuilder.

````Java
BigInteger hash = BigInteger.ZERO;
for (int x = 0; x < width; x++) {
	for (int y = 0; y < height; y++) {
		if (pixelValue[x][y] < compareAgainst) {
			hash = hash.shiftLeft(1);
		} else {
			hash = hash.shiftLeft(1).add(BigInteger.ONE);
		}
	}
}
return hash;
````
vs new
````Java
StringBuilder sb = new StringBuilder(hashResolution);
for (int x = 0; x < width; x++) {
	for (int y = 0; y < height; y++) {
		if (pixelValue[x][y] < compareAgainst) {
			sb.append(0);
		} else {
			sb.append(1);
		}
	}
}
return new BigInteger(sb.toString(),2);
````

In cooperation with #18

**Comment by KilianB** [comment]
for smaller than 64 bit hashes we could also fall back to long variables and do primitive bit shifting.

**Comment by KilianB** [comment]
jmh benchmarks.
<table>
<tr><th>Benchmark</th><th>Mode</th><th>Cnt</th><th>Score</th><th>Error</th><th>Units</th></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash32BigInteger</td><td>thrpt</td><td>25</td><td>152626,545</td><td>± 2741,163</td><td>ops/s</td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash32BigIntegerStringBuilder</td><td>thrpt</td><td>25</td><td>158690,519</td><td>± 2111,147</td><td>ops/s</td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash32MutableBigInteger</td><td>thrpt</td><td>25</td><td>157546,217</td><td>± 2026,677</td><td>ops/s</td></tr>
<tr><td colspan=6></td></tr>                                                            
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash128BigInteger</td><td>thrpt</td><td>25</td><td>104932,391</td><td>± 1387,338</td><td>ops/s</td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash128BigIntegerStringBuilder</td><td>thrpt</td><td>25</td><td>117392,216</td><td>± 1035,051</td><td>ops/s</td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash128MutableBigInteger</td><td>thrpt</td><td>25</td><td>116409,836</td><td>± 2436,276</td><td>ops/s</td></tr>
<tr><td colspan=6></td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash5000BigInteger</td><td>thrpt</td><td>25</td><td>1387,469</td><td>±  6,534</td><td>ops/s</td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash5000BigIntegerStringBuilder</td><td>thrpt</td><td>25</td><td>7690,280</td><td>± 96,870</td><td>  ops/s</td></tr>
<tr><td>kilianB.HashCreationBenchmark.benchmarkAverageHash5000MutableBigInteger</td><td>thrpt</td><td>25</td><td>7715,949</td><td>±   49,220</td><td>ops/s</td></tr>
</table>

Mutable big integer requires method handles and reflection access which is not warranted for the performance difference compare to the stringbuilder approach (as well as unit tests to guarantee that the implementation is working as expected).

A new StringBuilder is created for every hash creation. Caching a stringbuilder would result again in a performance gain at the cost of thread safety. In relation file IO is much more heavy and the treadoff isn't worth it. 

Migration to either stringbuilder or mutable big int allows to cut the first pass needed to estimate the hash length on an algorithm basis

**Comment by KilianB** [comment]
fixed with v 3.0.0
